File: core/src/main/java/com/shmedo/core/util/ActivityPool.java
package com.shmedo.core.util;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Stack;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2021/11/18 <br/>
 * 描述：     Activity管理池
 */
public class ActivityPool implements Application.ActivityLifecycleCallbacks {

    private static Stack<Activity> activityStack = new Stack<Activity>();

    public void init(Application app) {
        app.registerActivityLifecycleCallbacks(this);
    }

    @Override
    public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
        activityStack.add(activity);
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) {

    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {

    }

    @Override
    public void onActivityPaused(@NonNull Activity activity) {

    }

    @Override
    public void onActivityStopped(@NonNull Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {
        activityStack.remove(activity);
    }
}

File: core/src/main/java/com/shmedo/core/util/ActivityPool.kt
package com.shmedo.core.util

import android.app.Activity
import android.app.Application
import android.os.Bundle
import java.util.*

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2021/11/18 <br></br>
 * 描述：     Activity管理池
 */
class ActivityPool : Application.ActivityLifecycleCallbacks {
    fun init(app: Application) {
        app.registerActivityLifecycleCallbacks(this)
    }

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
        activityStack.add(activity)
    }

    override fun onActivityStarted(activity: Activity) {}
    override fun onActivityResumed(activity: Activity) {}
    override fun onActivityPaused(activity: Activity) {}
    override fun onActivityStopped(activity: Activity) {}
    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
    override fun onActivityDestroyed(activity: Activity) {
        activityStack.remove(activity)
    }

    companion object {
        private val activityStack = Stack<Activity>()
    }
}